### PR TITLE
test: refactor `B512Coder` tests

### PR DESCRIPTION
**Pull request by danielbate**
Related to #753

This PR will:
- Introduce a seperate test file for the service under test
- Seperate test cases for `encode` and `decode`
- Remove snapshot tests to bring expectations into the same test file
- Remove redundant tests to more clearly report on code coverage

### CODE REVIEW

**Review by arboleya — APPROVED** [review]
LGTM.

```
----------------|---------|----------|---------|---------|-------------------------------------
File            | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
----------------|---------|----------|---------|---------|-------------------------------------
  b512.ts       |   94.44 |    66.66 |     100 |   94.44 | 19
```

File: packages/abi-coder/src/coders/b512.test.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
+import B256Coder from './b512';
```
Comment (Torres-ssf) [inline packages/abi-coder/src/coders/b512.test.ts:1]:
```suggestion
import B512Coder from './b512';
```

File: packages/abi-coder/src/coders/b512.test.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    142, 157, 218, 111, 119, 147, 116, 90, 197, 170, 207, 158, 144, 124, 174, 48, 178, 160, 31, 223,
+    13, 35, 183, 117, 10, 133, 198, 164, 79, 202, 12, 41, 240, 144, 111, 157, 31, 30, 146, 230, 161,
+    251, 60, 61, 206, 243, 204, 59, 60, 219, 170, 226, 126, 71, 185, 217, 164, 198, 164, 252, 228,
+    207, 22, 178,
+  ]);
+  const B512_ZERO_DECODED =
+    '0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000';
+  const B512_ZERO_ENCODED = new Uint8Array(64);
+
+  const coder = new B256Coder();
```
Comment (Torres-ssf) [inline packages/abi-coder/src/coders/b512.test.ts:16]:
```suggestion
  const coder = new B512Coder();
```

**Review by Torres-ssf — CHANGES_REQUESTED** [review]
Great job :+1:

File: packages/abi-coder/src/coders/b512.test.ts (hunk 1 above)
Comment (arboleya) [reply packages/abi-coder/src/coders/b512.test.ts:1]: One of the (many) reasons I dislike default exports is that we can't enforce proper nomenclature to consumers or across our codebase, opening the doors to spelling errors and potentially different variable names on each import.

**Review by Torres-ssf — APPROVED** [review]

**Pull request merged**